vico/core/src/main/java/com/patrykandpatrick/vico/core/legend/HorizontalLegend.kt

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2023 by SaltedFish.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.patrykandpatrick.vico.core.legend
+
+import android.graphics.RectF
+import com.patrykandpatrick.vico.core.chart.draw.ChartDrawContext
+import com.patrykandpatrick.vico.core.component.dimension.Padding
+import com.patrykandpatrick.vico.core.component.text.HorizontalPosition
+import com.patrykandpatrick.vico.core.component.text.VerticalPosition
+import com.patrykandpatrick.vico.core.context.MeasureContext
+import com.patrykandpatrick.vico.core.dimensions.MutableDimensions
+import com.patrykandpatrick.vico.core.dimensions.emptyDimensions
+import com.patrykandpatrick.vico.core.extension.half
+import com.patrykandpatrick.vico.core.legend.VerticalLegend.Item
+
+/**
+ * [HorizontalLegend] displays legend items in a line wrapping horizontal row.
+ *
+ * @param items a [Collection] of [Item]s to be displayed by this [HorizontalLegend].
+ * @param iconSizeDp defines the size of all [Item.icon]s.
+ * @param iconPaddingDp defines the padding between each [Item.icon] and its corresponding [Item.label].
+ * @param lineSpacingDp define the vertical spacing between lines.
+ * @param spacingDp defines the horizon spacing between each [Item] in line.
+ * @param padding defines the padding of the content.
+ */
+public open class HorizontalLegend(
+    public var items: Collection<Item>,
+    public var iconSizeDp: Float,
+    public var iconPaddingDp: Float,
+    public var lineSpacingDp: Float = 0f,
+    public var spacingDp: Float = 0f,
+    override val padding: MutableDimensions = emptyDimensions(),
+) : Legend, Padding {
+    // Cache the height of each line
+    private val heights = mutableListOf<Float>()
+
+    // Cache the items in each line
+    private val lines = mutableListOf<MutableList<Item>>(mutableListOf())
```

Comment:
We generally avoid adding comments to code that is self-explanatory. Also, to keep the code style consistent, there should be a new line added after the definition.
```suggestion
    private val lines = mutableListOf<MutableList<Item>>(mutableListOf())
    
```